homeassistant/components/yale_smart_alarm/binary_sensor.py

```diff
@@ -0,0 +1,44 @@
+"""Binary sensors for Yale Alarm."""
+from __future__ import annotations
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import COORDINATOR, DOMAIN
+from .coordinator import YaleDataUpdateCoordinator
+from .entity import YaleEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Yale binary sensor entry."""
+
+    coordinator: YaleDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        COORDINATOR
+    ]
+
+    async_add_entities(
+        YaleBinarySensor(coordinator, data) for data in coordinator.data["door_windows"]
+    )
+
+
+class YaleBinarySensor(YaleEntity, BinarySensorEntity):
+    """Representation of a Yale binary sensor."""
+
+    _attr_device_class = BinarySensorDeviceClass.DOOR
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+        for contact in self.coordinator.data["door_windows"]:
+            return bool(
+                contact["address"] == self._attr_unique_id
+                and contact["_state"] == "open"
+            )
```

Comment:
Can we make a `dict` when the coordinator updates with {contact["address"]: contact["_state"]} so we don't have to do a linear search for each sensor and can do something like?

return bool(sensor_map[self._attr_unique_id] == "open") ?